src/api/pipelineRuns.js

```diff
@@ -170,30 +173,46 @@ export function createPipelineRun({
   return post(uri, payload).then(({ body }) => body);
 }
 
-export function rerunPipelineRun(pipelineRun) {
-  const { annotations, labels, name, namespace } = pipelineRun.metadata;
+export function generateNewPipelineRunPayload(pipelineRun, rerun) {
+  const { annotations, labels, name, namespace, generateName } =
+    pipelineRun.metadata;
 
   const payload = deepClone(pipelineRun);
   payload.apiVersion =
     payload.apiVersion || `tekton.dev/${getTektonPipelinesAPIVersion()}`;
   payload.kind = payload.kind || 'PipelineRun';
+
+  function getGenerateName() {
+    if (rerun) {
+      return getGenerateNamePrefixForRerun(name);
+    }
+
+    return generateName || getGenerateNamePrefixForNewRun(name);
+  }
+
   payload.metadata = {
-    annotations: annotations || {},
-    generateName: getGenerateNamePrefixForRerun(name),
-    labels: {
-      ...labels,
-      'dashboard.tekton.dev/rerunOf': name
-    },
+    annotations,
+    generateName: getGenerateName(),
+    labels,
     namespace
   };
+  if (rerun) {
+    payload.metadata.labels = {
+      ...labels,
+      'dashboard.tekton.dev/rerunOf': name
+    };
+  }
 
-  Object.keys(payload.metadata.labels).forEach(label => {
-    if (label.startsWith('tekton.dev/')) {
-      delete payload.metadata.labels[label];
-    }
-  });
+  if (payload.metadata.labels) {
```

Comment:
with the suggestion above this check wouldn't be required as `labels` would always be defined

similar below for annotations